### PR TITLE
Add Azure.Core to solutions that have TestFramework

### PR DESCRIPTION
**Pull request by pakrym**
The TestFramework has a dependency on Azure.Core project. The build fails in Visual Studio if the project is not part of the solution.

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/perf/Azure.Messaging.EventHubs.Processor.Perf.sln [hunk 1]
```diff
@@ -15,6 +15,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Azure.Core.TestFramework",
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Azure.Test.Perf", "..\..\..\..\common\Perf\Azure.Test.Perf\Azure.Test.Perf.csproj", "{1F5EE4F0-7DE8-4E3E-819C-58F34D17C2A1}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Azure.Core", "..\..\..\core\Azure.Core\src\Azure.Core.csproj", "{01AA6564-166A-411E-9B0D-6A2CABA246F7}"
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/perf/Azure.Messaging.EventHubs.Processor.Perf.sln:18]: Do you mind deleting `Azure.Messaging.EventHubs.Perf.sln`?   It's no longer needed and I missed doing so when I moved to the [filter set](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs/perf/Azure.Messaging.EventHubs.Perf.slnf) the other day.

**Review by christothes — APPROVED** [review]

**Review by jsquire — APPROVED** [review]

**Review by annelo-msft — APPROVED** [review]

**Comment by annelo-msft** [comment]
Is there any drawback to keeping the project reference in these solutions indefinitely?

**Comment by pakrym** [comment]
>Is there any drawback to keeping the project references in these solutions indefinitely?

Might be misleading to partners. They might think that it's possible to debug Azure.Core because it's part of the solution but it's not always true.
Also, you pay the build cost every time even when Azure.Core is coming from a package reference.

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/perf/Azure.Messaging.EventHubs.Processor.Perf.sln (hunk 1 above)
Comment (pakrym) [reply sdk/eventhub/Azure.Messaging.EventHubs.Processor/perf/Azure.Messaging.EventHubs.Processor.Perf.sln:18]: Sorry, this PR is large with a ton of checks. Don't want to kick them all again.

**Comment by annelo-msft** [comment]
Thanks!  So the plan here is to remove these once Core is next released.

**Comment by JoshLove-msft** [comment]
> Thanks! So the plan here is to remove these once Core is next released.

Are we switching Test Framework to have a package reference on Core?

**Comment by annelo-msft** [comment]
@JoshLove-msft, it did before I made this change: https://github.com/Azure/azure-sdk-for-net/commit/fd9ad862e0d36da4173d2a9dc078b9e06ed8863b#diff-63fb8c0cdfa41932197740d9eb7e1702d3c6e50385afdb2008a24a4543c36661R7.  IIRC, we usually move projects that rely on new Core features to project references until the new package is released and then move them back to package references.

**Comment by JoshLove-msft** [comment]
> @JoshLove-msft, it did before I made this change: [fd9ad86#diff-63fb8c0cdfa41932197740d9eb7e1702d3c6e50385afdb2008a24a4543c36661R7](https://github.com/Azure/azure-sdk-for-net/commit/fd9ad862e0d36da4173d2a9dc078b9e06ed8863b#diff-63fb8c0cdfa41932197740d9eb7e1702d3c6e50385afdb2008a24a4543c36661R7). IIRC, we usually move projects that rely on new Core features to project references until the new package is released and then move them back to package references.

Ah I hadn't realized that it was using package reference before. I thought we always had a proj ref there.

**Pull request merged**